### PR TITLE
Keep syspurpose synced between client and server side for both subman and syspurpose tools

### PR DESCRIPTION
**Pull request by cnsnyder**
This PR needs a squash and some test fixes, putting here to get review in parallel with on-going test fixes.

### CODE REVIEW

**Comment by cnsnyder** [comment]
Tests should pass finally.

File: src/subscription_manager/managerlib.py [hunk 1]
```diff
@@ -856,7 +856,7 @@ def clean_all_data(backup=True):
     # for deleting persistent caches
     cache.ProfileManager.delete_cache()
     cache.InstalledProductsManager.delete_cache()
-    cache.SyspurposeCache.delete_cache()
+    # TODO Clear the syspurpose cache if unregistered.
```
Comment (kahowell) [inline src/subscription_manager/managerlib.py]: Going to implement this TODO now or later? What is the effect of not implementing now?

File: src/subscription_manager/managercli.py [hunk 2]
```diff
@@ -1088,8 +1109,12 @@ def _do_command(self):
             else:
                 self.show_service_level()
 
-    def _set(self, service_level):
-        save_sla_to_syspurpose_metadata(service_level)
+    def set(self):
+        if self.cp.has_capability("syspurpose"):
+            super(ServiceLevelCommand, self).set()
+        else:
+            # TODO Find old impl of service level command.
```
Comment (kahowell) [inline src/subscription_manager/managercli.py]: impl now or later?

Comment (cnsnyder) [reply src/subscription_manager/managercli.py]: Later, need it fixed for next build.

File: src/subscription_manager/managerlib.py (hunk 1 above)
Comment (cnsnyder) [reply src/subscription_manager/managerlib.py]: Fixed

**Comment by cnsnyder** [comment]
@candlepin-pull-request-bot Please retest this

**Comment by kahowell** [comment]
After some discussion, we will hold off on removing try, imports for another later PR.

**Comment by cnsnyder** [comment]
@candlepin-pull-request-bot Please retest this

**Comment by cnsnyder** [comment]
@candlepin-pull-request-bot Please retest this

**Pull request merged**